public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataQueriesTab.tsx

```diff
@@ -92,16 +101,65 @@ export class PanelDataQueriesTab extends SceneObjectBase<PanelDataQueriesTabStat
 
   onQueriesChange = (queries: DataQuery[]) => {
     this._panelManager.changeQueries(queries);
+
+    // this so a rerender is triggered and counter for the tab is updated
+    if (this._parent instanceof PanelDataPane) {
```

Comment:
I tried a couple of approaches here, and i found that this one works and also using `forceRender` on the parent. There's probably a better approach to have the parent rerender and update the counter. I tried subscribing to `panelManager` state in the parent, but that didn't work, so I'm not sure what else can be done here.

Reply:
Have an idea, will open a proposal in the morning

Reply:
@harisrozajac - i think we can do it this way: https://github.com/grafana/grafana/pull/81147

Feel free to cherry pic that commit. 

cc @oscarkilhed  - i think this is related to your [transformations PR](https://github.com/grafana/grafana/pull/80738) as well

Reply:
Cool, yeah works on my branch as well. Suggest we cherry-pick the change to only one of these PRs to avoid unnecessary merge headaches.

Reply:
I went ahead and merged my PR without the change. I'll circle back and fix the counter when this change is merged.

Reply:
i'm still having some issues with making tests work so i would say let's merge this one without tests so @oscarkilhed can be unblocked for transformations